src/couch/priv/couch_js/60/util.cpp

```diff
@@ -231,7 +212,7 @@ couch_readline(JSContext* cx, FILE* fp)
     bytes = tmp;
     byteslen = used;
 
-    str = string_to_js(cx, std::string(tmp));
+    str = string_to_js(cx, std::string(tmp, byteslen));
```

Comment:
Why not replace this with the direct call to `dec_string` and drop `string_to_js`? Unless you think it'd be useful over in `utf8.cpp`/`http.cpp` (for the short time left that we care about those)?

I wonder if we can't replace some of this with something native now.

At the very least, `dec_char` has an equivalent in [`Utf8ToOneUcs4Char`](https://searchfox.org/mozilla-central/source/js/src/vm/CharacterEncoding.cpp#178) (in `js/CharacterEncoding.h`.

This is used in [`Decode`](https://searchfox.org/mozilla-central/source/js/src/builtin/String.cpp#4219) but unfortunately this and its wrappers are not public anywhere. On the whole this doesn't look too different to our `dec_charbuf` or `dec_string`, but it would be useful to compare and ensure our code is current vs. what Firefox does.

I'll take this on shortly for the SM68 port if you want to punt on it for this PR.

Reply:
I've gone through and repalced and removed utf8.{h,c} so this isn't just a pass through shim any more.

Reply:
Great, marking as resolved. I'll pull over your changes into my SM68 PR once you've merged this, then.